collect_app/src/main/java/org/odk/collect/android/utilities/ClientHttpConnection.java

```diff
@@ -135,6 +146,64 @@ public InputStream getInputStream(@NonNull URI uri, String contentType, final in
         return downloadStream;
     }
 
+    @Override
+    public int httpHeadRequest(@NonNull URI uri, Map<String, String> responseHeaders) throws Exception {
```

Comment:
I see the same flaw here as in https://github.com/opendatakit/collect/pull/2257/commits/1805c962ef6f9befc0bea51a23985564820c4030#r202655171

The real output of this method would be the HTTP status code and a map of headers. We could return an object with that instead of having an output arg.

Reply:
Nevermind! I see you've done that in other commits :)

Reply:
Agreed, I'll update accordingly

Reply:
Awesome!